TenzDelek/DearDiary
# Feature/add husky Pre-commit

## Changes
- Added Husky pre-commit hooks to run linting and tests before each commit

## Reason
To improve code quality and catch issues early in the development process

## Implementation Details
- Installed Husky as a dev dependency
- Set up a pre-commit hook to run `npm run lint` and `npm test`

## How to Test
1. Make a change to any file
2. Attempt to commit the change
3. Verify that linting and tests run before the commit is allowed
